frontends/common/parser_options.cpp

```diff
@@ -455,19 +476,25 @@ std::optional<ParserOptions::PreprocessorResult> ParserOptions::preprocess() con
         }
         return std::nullopt;
     }
-    return ParserOptions::PreprocessorResult(in, &closeFile);
-}
 
-// From (folder, file.ext, suffix)  returns
-// folder/file-suffix.ext
-static std::filesystem::path makeFileName(const std::filesystem::path &folder,
-                                          const std::filesystem::path &name,
-                                          std::string_view baseSuffix) {
-    std::filesystem::path newName(name.stem());
-    newName += baseSuffix;
-    newName += name.extension();
+    if (savePreprocessed) {
+        std::filesystem::path fileName = makeFileName(dumpFolder, "repro.p4", "");
```

Comment:
I think that using `<program_name>.p4pp` would be better, compared to having a fairly arbitrary fixed name.